### PR TITLE
Remove the disabled autogenerated id optimization from InternalEngine

### PR DESCRIPTION
**Pull request by bleskes**
If a document is indexed into ES with no id, ES will generate one for it. We used to have an optimization for this case where the engine will not try to resolve the ids of these request in the existing index but immediately try to index them. This optimization has proven to be the source of brittle bugs (solved!) and we disabled it in 1.5, preparing for it to be removed if no performance degradation was found. Since we haven't seen any such degradation we can remove it.

Along with the removal of the optmization, we can remove the autogenerate id flag on indexing requests and the can have duplicate flag. The only downside of the removal of the canHaveDuplicate flag is that we can't make sure any more that when we retry an autogenerated id create operation we will ignore any document already exists exception (See https://github.com/elasticsearch/elasticsearch/pull/9125  for background and discussion). To work around this, we don't set the operation to CREATE any more when we generate an id, so the resulting request will never fail when it finds an existing doc but do return a version of 2. I think that's acceptable.

### CODE REVIEW

**Comment by bleskes** [comment]
@s1monw can you take a look?

File: core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java [hunk 1]
```diff
@@ -42,7 +42,10 @@
 import org.elasticsearch.cluster.metadata.IndexMetaData;
 import org.elasticsearch.cluster.metadata.IndexNameExpressionResolver;
 import org.elasticsearch.cluster.node.DiscoveryNode;
-import org.elasticsearch.cluster.routing.*;
+import org.elasticsearch.cluster.routing.IndexRoutingTable;
+import org.elasticsearch.cluster.routing.IndexShardRoutingTable;
+import org.elasticsearch.cluster.routing.ShardIterator;
+import org.elasticsearch.cluster.routing.ShardRouting;
 import org.elasticsearch.common.Nullable;
```
Comment (mikemccand) [inline core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java]: Maybe we should standardize this code style so we all don't have to keep seeing silly import changes :)

**Comment by mikemccand** [comment]
LGTM

File: core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java (hunk 1 above)
Comment (bleskes) [inline core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java]: yeah, long time ago the decision was to go with intellij defaults (with one exception)... I personally don't mind to go with whatever - as long as we agree :)

Comment (mikemccand) [inline core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java]: >  I personally don't mind to go with whatever - as long as we agree :)

Same here!

**Pull request closed without merging**